datafusion/common/src/hash_utils.rs

```diff
@@ -535,6 +580,23 @@ fn hash_single_array(
             let array = as_union_array(array)?;
             hash_union_array(array, random_state, hashes_buffer)?;
         }
+        DataType::RunEndEncoded(run_ends_type, _) => {
+            match run_ends_type.data_type() {
+                DataType::Int16 => {
+                    let array = as_run_array::<Int16Type>(array)?;
+                    hash_run_array(array, random_state, hashes_buffer, rehash)?;
+                }
+                DataType::Int32 => {
+                    let array = as_run_array::<Int32Type>(array)?;
+                    hash_run_array(array, random_state, hashes_buffer, rehash)?;
+                }
+                DataType::Int64 => {
+                    let array = as_run_array::<Int64Type>(array)?;
+                    hash_run_array(array, random_state, hashes_buffer, rehash)?;
+                }
+                _ => unreachable!("RunEndEncoded must have Int16, Int32, or Int64 run ends")
+            }
+        }
```

Comment:
Perhaps can use [`downcast_run_array`](https://docs.rs/datafusion/latest/datafusion/common/arrow/array/macro.downcast_run_array.html) similar to dictionary above